lexicon/providers/njalla.py

```diff
@@ -0,0 +1,145 @@
+        params = {
+            'domain': self.domain,
+        }
+        result = self._api_call('list-records', params)
+
+        records = result['records']
+        processed_records = [{
+            'id': record['id'],
+            'type': record['type'],
+            'name': self._full_name(record['name']),
+            'ttl': record['ttl'],
+            'content': record['content'],
+        } for record in records]
+        filtered_records = [record for record in processed_records if (
+            (rtype is None or record['type'] == rtype) and
+            (name is None or record['name'] == self._full_name(name)) and
+            (content is None or record['content'] == content))]
+
+        LOGGER.debug('list_records: %s', filtered_records)
+        return filtered_records
+
+    # Create or update a record.
+    def _update_record(self, identifier, rtype=None, name=None, content=None):
+        if not identifier:
+            identifier = self._get_record_identifier(rtype=rtype, name=name)
+
+        params = {
+            'id': identifier,
+            'domain': self.domain,
+            'content': content,
+        }
+        result = self._api_call('edit-record', params)
+
+        LOGGER.debug('update_record: %s', result)
+        return result
+
+    # Delete an existing record.
+    # If record does not exist, do nothing.
+    def _delete_record(self, identifier=None, rtype=None, name=None, content=None):
+        if not identifier:
+            identifier = self._get_record_identifier(rtype=rtype, name=name, content=content)
+
+        params = {
+            'domain': self.domain,
+            'id': identifier,
+        }
+        self._api_call('remove-record', params)
+
+        LOGGER.debug('delete_record: %s', True)
+        return True
+
+    # Helpers
+    def _api_call(self, method, params):
+        if self._get_provider_option('auth_token') is None:
+            raise Exception('Must provide API token')
+
+        token = self._get_provider_option('auth_token')
+        headers = {
+            'Authorization': 'Njalla ' + token,
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+        }
+        data = {
+            'method': method,
+            'params': params,
+        }
+        response = requests.post(self.api_endpoint,
+                                 headers=headers,
+                                 json=data).json()
+
+        if 'error' in response.keys():
+            error = response['error']
+            raise Exception('%d: %s' % (error['code'], error['message']))
+
+        return response['result']
+
+    def _get_record_identifier(self, rtype=None, name=None, content=None):
+        records = self._list_records(rtype=rtype, name=name, content=content)
+        if len(records) == 1:
+            return records[0]['id']
+
+        raise Exception('Unambiguous record could not be found.')
```

Comment:
Just to inform you here. You have basically two choices if more than one record is found:
* error out, like you did
* get the first matching record, and warn the user that more than a record was found, and so the first one will be taken

Most of the providers use the second option, but it is your call.

Reply:
I'm using this method in `_delete_record` so deleting the first of multiple matched records seems dangerous.

Reply:
I agree. My option makes more sense in update operation, but for deleting it is bad.